test/test_torch.py

```diff
@@ -8528,29 +8528,23 @@ def test_cartesian_prod(self):
         a = torch.tensor(1)
         b = torch.tensor([1, 2, 3])
         c = torch.tensor([1, 2])
-        a_, b_, c_ = torch.cartesian_prod([a, b, c])
-        expected_b = torch.tensor([1, 1, 2, 2, 3, 3])
-        expected_c = torch.tensor([1, 2, 1, 2, 1, 2])
-        self.assertEqual((a - a_).abs().max().item(), 0)
-        self.assertEqual(b_, expected_b)
-        self.assertEqual(c_, expected_c)
+        prod = torch.cartesian_prod([a, b, c])
+        expected = torch.tensor(list(product([a], b, c)))
+        self.assertEqual(expected, prod)
```

Comment:
Please add some tests for edge cases. In particular, I'd like to see the following:
- one tensor is an empty tensor `torch.zeros(0, dtype=torch.int)`
- what happens when you call cartesian_prod with only one element?

Reply:
@zou3519 Sorry I forgot to add the "call cartesian_prod with only one element" case. I just added this case, as well as `r=1` for combinations.

Should `cartesian_prod("tensor of shape (3,)")` result a tensor of shape `(3,)` or `(3,1)`? Currently it is the latter.

Reply:
It should return a tensor of shape (3,), right? If you cartesian product tensors of size `(3,)` and `(2,)`, the result should have shape `(3, 2)`, right?